crates/sui-bridge-indexer/src/sui_bridge_indexer.rs

```diff
@@ -0,0 +1,286 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{anyhow, Error};
+use async_trait::async_trait;
+use diesel::dsl::now;
+use diesel::{Connection, OptionalExtension, QueryDsl, RunQueryDsl, SelectableHelper};
+use diesel::{ExpressionMethods, TextExpressionMethods};
+use tracing::info;
+
+use sui_bridge::events::{
+    MoveTokenDepositedEvent, MoveTokenTransferApproved, MoveTokenTransferClaimed,
+};
+use sui_types::effects::TransactionEffectsAPI;
+use sui_types::event::Event;
+use sui_types::execution_status::ExecutionStatus;
+use sui_types::full_checkpoint_content::CheckpointTransaction;
+use sui_types::{BRIDGE_ADDRESS, SUI_BRIDGE_OBJECT_ID};
+
+use crate::indexer_builder::{CheckpointTxnData, DataMapper, IndexerProgressStore, Persistent};
+use crate::metrics::BridgeIndexerMetrics;
+use crate::postgres_manager::PgPool;
+use crate::schema::progress_store::{columns, dsl};
+use crate::schema::{sui_error_transactions, token_transfer, token_transfer_data};
+use crate::sui_checkpoint_ingestion::Task;
+use crate::{
+    models, schema, BridgeDataSource, ProcessedTxnData, SuiTxnError, TokenTransfer,
+    TokenTransferData, TokenTransferStatus,
+};
+
+/// Persistent layer impl
+#[derive(Clone)]
+pub struct PgBridgePersistent {
+    pool: PgPool,
+}
+
+impl PgBridgePersistent {
+    pub fn new(pool: PgPool) -> Self {
+        Self { pool }
+    }
+}
+
+impl Persistent<ProcessedTxnData> for PgBridgePersistent {
+    fn write(&self, data: Vec<ProcessedTxnData>) -> Result<(), Error> {
+        if data.is_empty() {
```

Comment:
this should not be in `*sui*_bridge_indexer.rs` file, right? it's shared between eth and sui

Reply:
will do another house keeping PR to move things around

Reply:
can we plz add a TODO here